tests/unit/form/Validator.spec.ts

```diff
@@ -0,0 +1,95 @@
+import FormValues from '@/form/FormValues';
+import Validator, { ValidationResult } from '@/form/Validator';
+
+describe( 'validator', () => {
+	it( 'full form', () => {
+		const formValues: FormValues = {
+			property: {
+				id: 'P31',
+				label: 'instance of',
+			},
+			value: 'Q5',
+		};
+
+		const expectedResult: ValidationResult = {
+			formErrors: [],
+			fieldErrors: {
+				property: null,
+				value: null,
+			},
+		};
+
+		const validator = new Validator( formValues );
+
+		expect( validator.validate() ).toStrictEqual( expectedResult );
+	} );
+
+	it( 'property missing', () => {
+		const formValues: FormValues = {
+			property: null,
+			value: 'Q5',
+		};
+
+		const formErrors = [
+			{
+				type: 'error',
+			},
+		];
+
+		const validator = new Validator( formValues );
+		const validationResult = validator.validate();
+
+		expect( validationResult.formErrors ).toHaveLength( 1 );
+		expect( validationResult.formErrors[ 0 ].type ).toBe( formErrors[ 0 ].type );
```

Comment:
I think it makes more sense to construct the actual error objects here, with the literal message for now, and to assert them with `.toStrictEqual`. Then those 5 assertions here can be replaced with two.

We can later replace the message with the message key when the i18n story is done.

Reply:
That's exactly why I did five asserts instead of one. It's not just i18n itself, it's also repeating the error message over and over, meaning if we want to change the message (even a missing dot), we need to change lots of tests and I don't like such tight binding.

Reply:
Mh, I can accept that. But then I'll add a AC to the i18n ticket to fix these tests when we have the i18n implemented.

Reply:
=> https://phabricator.wikimedia.org/T268014

Reply:
Thanks!